src/main/typescript/hooks/renderChatMessage/decorateSystemMessage/index.ts

```diff
@@ -23,6 +24,8 @@ export default function decorateSystemMessage(
     case "painThreshold":
       decoratePTMessage(systemFlags as PainThresholdFlags, html);
       break;
+    case "roll":
+      decorateCheck(systemFlags as CheckFlags, html);
```

Comment:
Can you check if those `as` casts are still necessary or if we have some more safe way to narrow this down, please?